### PR TITLE
feat(signer): Add pocket-ic bindings for the cycles ledger

### PR DESCRIPTION
**Pull request by bitdivine**
# Motivation
In tests, we will need to call the cycles ledger.

# Changes
- Add support for zero or one arguments to the handlebars rust binding template.  Currently only one argument is supported and some ledger functions take no arguments.
- Add bindings to the cycles ledger

# Tests
There are existing tests that the bindings are up to date and that the rust code compiles.  But the code is not used yet in any way; it will be used in tests.

### CODE REVIEW

File: scripts/bind/pic/cycles_depositor.toml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [didc.rust]
 target = "custom"
-template = "scripts/bind/pic/cycles_depositor.hbs"
+template = "scripts/bind/pic/template.hbs"
```
Comment (bitdivine) [inline scripts/bind/pic/cycles_depositor.toml:3]: We re-use the template, so the template needs to be renamed accordingly.

File: scripts/bind/pic/template.hbs [hunk 2]
```diff
@@ -7,8 +7,11 @@ use pocket_ic::PocketIc;
 
 use crate::utils::pic_canister::{PicCanister, PicCanisterTrait};
 
-{{type_defs}}
+{{#if submodule}}
+pub mod {{submodule}};
+{{/if}}
```
Comment (bitdivine) [inline scripts/bind/pic/template.hbs:12]: This optionally adds a submodule, that can be used to add methods to the auto-generated types.  I would prefer this to be an array but the didc toml only supports strings as template vars.  And de-facto one submodule is plenty for our use case.

File: scripts/bind/pic/template.hbs [hunk 3]
```diff
@@ -39,7 +42,13 @@ impl PicCanisterTrait for {{PascalCase canister_name}}Pic {
...
   {{#each methods}}
   pub fn {{this.name}}(&self, _caller: Principal{{#each this.args}}, {{this.0}}: &{{this.1}}{{/each}}) -> Result<{{vec_to_arity this.rets}}, String> {
-      self.update(self.canister_id, "{{escape_debug this.original_name}}", arg0)
+      {{#if this.args.[1]}}
+      unimplemented!("The update function handles only a single argument.")
+      {{else if this.args.[0]}}
+      self.update(self.canister_id, "{{escape_debug this.original_name}}", {{#each this.args}}{{this.0}}{{/each}})
+      {{else}}
+      self.update(self.canister_id, "{{escape_debug this.original_name}}", ())
+      {{/if}}
```
Comment (bitdivine) [inline scripts/bind/pic/template.hbs:51]: The template on the left supports only a single argument called `arg0`.  On the right, we support zero arguments, one argument with any name, but not (yet) more arguments.  Supporting more arguments will be needed, but requires updating the `update()` function, and I don't want to make this PR too big.

File: src/signer/canister/tests/it/canister/cycles_ledger/impls.rs [hunk 4]
```diff
@@ -0,0 +1,19 @@
+use candid::Nat;
+
+use super::{Account, ApproveArgs};
+
+impl ApproveArgs {
+    /// Simple approval arguments.
+    pub fn new(spender: Account, amount: Nat) -> Self {
```
Comment (bitdivine) [inline src/signer/canister/tests/it/canister/cycles_ledger/impls.rs:7]: Most of the time, ICRC2 approvals populate only these two values.  In our tests anyway.  Probably in real life as well.

File: src/signer/canister/tests/it/canister/cycles_ledger.rs [hunk 5]
```diff
@@ -0,0 +1,583 @@
+//! Bindings to the cycles_depositor canister, generated by ./scripts/bind/pic/cycles_ledger.sh
```
Comment (bitdivine) [inline src/signer/canister/tests/it/canister/cycles_ledger.rs:1]: This autogenerated file is the main contribution of this PR.

**Review by lmuntaner — APPROVED** [review]
LGTM!

**Pull request merged**